src/middlewared/middlewared/plugins/system/product.py

```diff
@@ -20,6 +20,11 @@
 PRODUCT_NAME = 'TrueNAS'
 
 
+class ProductType():
```

Comment:
This is a global mutable type. We can also reduce the size of this object considerably.

We can do 1 of 2 things:
Option 1:
```
from dataclasses import dataclass

@dataclass(slots=True, frozen=True)
class ProductType:
    SCALE: str = 'SCALE'
    SCALE_ENTERPRISE: str = 'SCALE_ENTERPRISE'
```

Option 2:
```
from enum import Enum

class ProductType(str, Enum):
    SCALE = 'SCALE'
    SCALE_ENTERPRISE = 'SCALE_ENTERPRISE'
```

For option number 1, we reduce memory size considerably and we make it immutable.

For option number 2, we reduce memory size to 0 however you need to change all the places we reference the enum to do something like this `ProductType.SCALE.value`. If you don't add the `.value` at the end and only use `ProductType.SCALE` it creates a _new_ instance of the enum class which is an expensive operation.

Reply:
I've got no particular preference.  Which option do _you_ prefer?